_posts/2019-05-09-How-We-Release-Code.md

```diff
@@ -0,0 +1,96 @@
+
+1. Can the project be sustainable?
+2. Do Zalando have any value in taking ownership?
+3. Does it have any value to anyone outside Zalando?
+
+When source code is released as open source, you are essentially giving away something of value, you are also taking responsibility for spending time and resources on the additional overhead associated with open sourcing, this loss of value and use of resources should be justified, there are multiple ways to look at this, such as:
+
+1. The project contributes positively to the employer branding efforts and supports hiring of tech talent
+2. The project helps establish the company as a leader in a certain domain
+3. The project will gain features and bugfixes from external community members
+4. The maintainer team could gain valuable knowledge through collaborating with external community members
+
+For Zalando we've seen several projects contribute to our employer branding efforts, it is however a byproduct and should not be the main reason for open sourcing. It is ofcourse nice that Zalando is recognised for its Kubernetes ([External-Dns](https://github.com/kubernetes-incubator/external-dns), [Stackset-Controller](https://github.com/zalando-incubator/stackset-controller), [es-operator](https://github.com/zalando-incubator/es-operator), PostGreSQL ([Patroni](https://github.com/zalando/patroni) and [postgres-operator](https://github.com/zalando/postgres-operator) and Machine Learning projects ([Flair](https://github.com/zalandoresearch/flair) and [Fashion-mnist](https://github.com/zalandoresearch/fashion-mnist), but it is a value which is hard to measure the effect of and not a long-term motivator for neither the maintainer or company. 
+
+### Reviwing project quality
+Besides discussing the WHY, the open source team looks at compliance-specific areas which could be a blocker for releasing:
+
+1. Do we use dependencies which have incompatible licensing
+2. Does the source code contain anything confidential (such as tokens, urls, passwords, etc) 
+3. Does the project contain functionality or IP which gives Zalando a competetive advantage (such as the code that powers our search results)
+4. Is the project something Zalando would consider trying to patent?
+
+We use a dependency licensing scanning tool, as well as a source code scanner to look for tokens and passwords, to automate this as much as possible.
+
+### Review Meeting
+Once a month the review group sits down with the maintainers proposing new projects, here the dicussions from the mailing-list will be concluded and a release decision is made. The project is either granted release, rejected, or the maintainers are asked to improve certain aspects of the project before it can be released, depending on the number of proposals these decisions are reached in 30-60 minutes and release meetings are setup to formally prepare the files and publish them on Github.com.
```

Comment:
```suggestion
Once a month the review group sits down with the maintainers proposing new projects. Here, the dicussions from the mailing-list will be concluded and a release decision is made. The project is either granted release, rejected, or the maintainers are asked to improve certain aspects of the project before it can be released. Depending on the number of proposals, these decisions are reached in 30-60 minutes and release meetings are set up to formally prepare the files and publish them on Github.com.
```